File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/location/OnRenderModeChangedListener.java
package org.maplibre.android.location;

import org.maplibre.android.location.modes.RenderMode;

/**
 * Listener that gets invoked when layer render mode changes.
 */
public interface OnRenderModeChangedListener {

  /**
   * Invoked on every {@link RenderMode} change.
   *
   * @param currentMode current active {@link RenderMode}.
   */
  void onRenderModeChanged(@RenderMode.Mode int currentMode);
}

File: platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/location/OnRenderModeChangedListener.kt
package org.maplibre.android.location

import org.maplibre.android.location.modes.RenderMode

/**
 * Listener that gets invoked when layer render mode changes.
 */
fun interface OnRenderModeChangedListener {
    /**
     * Invoked on every [RenderMode] change.
     *
     * @param currentMode current active [RenderMode].
     */
    fun onRenderModeChanged(
        @RenderMode.Mode currentMode: Int,
    )
}
